src/lane.cpp

```diff
@@ -1,28 +1,68 @@
 // Header
 #include "lane.hpp"
 
-// stlib
-#include <vector>
-#include <algorithm>
-
-// Releases all graphics resources
-bool Lane::init(float world_scale)
+Lane::Lane(float world_scale, direction dir)
 {
-	m_cars = new Car[MaxCarsPerLane];
-	return true;
+	m_world_scale = world_scale;
+	m_dir = dir;
+	m_time_remaining = m_max_time_per_car;
 }
 
-void Lane::destroy()
+Lane::~Lane()
 {
-	delete m_cars;
+	m_cars.clear();
 }
 
 float Lane::get_time_remaining() const
 {
     return m_time_remaining;
 }
 
-Car* Lane::get_cars() const
+std::vector<Car> Lane::get_cars() const
 {
     return m_cars;
 }
+
+bool Lane::update(float ms)
+{
+	m_time_remaining -= ms;
+	if (m_time_remaining <= 0)
+	{
+		this->turn_car();
+		m_time_remaining = m_max_time_per_car;
+	}
+	return true;
+}
+
+void Lane::add_car()
+{
+	if (m_cars.size() < MaxCarsPerLane) {
+		Car new_car;
+		if(new_car.init(m_world_scale)){
+			m_cars.emplace_back(new_car);
+			//new_car.enter_lane(direction dir); <-- function to animate moving car new up to previous car in line
+		}
+	} else {
+		//throw error? How do I do that in C++...

```

Comment:
Probably don't need to throw an error unless it's for debugging. If we can't add a car, then that's fine? 

Will look something like this: 
#include <stdexcept>

int compare( int a, int b ) {
    if ( a < 0 || b < 0 ) {
        throw std::invalid_argument( "received negative value" );
    }
}
https://stackoverflow.com/questions/8480640/how-to-throw-a-c-exception

Reply:
will throwing an error cause the game to crash or will it just print out a message in the terminal? I think the most we would want to do is the second one, so an actual error is probably overkill. I might add a print statement for debugging purposes, but you're right that it's probably unnecessary